easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,210 @@
+            elif flag == 'avx512f':
+                if getattr(self, 'avx512f'):
+                    self.cfg.update('configopts', '--enable-avx512')
+                else:
+                    self.cfg.update('configopts', '--disable-avx512')
+            else:
+                if getattr(self, flag):
+                    self.cfg.update('configopts', '--enable-%s' % flag)
+                else:
+                    self.cfg.update('configopts', '--disable-%s' % flag)
+
+        # By default ELPA tries to use MPI and configure fails if it's not available
+        # so we turn off MPI support unless MPI support is requested via the usempi toolchain option.
+        # We also set the LIBS environmet variable to detect the correct linalg library
+        # depending on the MPI availability.
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', 'LIBS="$LIBSCALAPACK"')
+        else:
+            self.cfg.update('configopts', '--with-mpi=no')
+            self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
+
+        # make all builds verbose
+        self.cfg.update('buildopts', 'V=1')
+
+        # keep track of common configopts specified in easyconfig file,
+        # so we can include them in each iteration later
+        common_config_opts = self.cfg['configopts']
+
+        self.cfg['configopts'] = []
+
+        self.cfg.update('configopts', ['--disable-openmp ' + common_config_opts])
+        if self.toolchain.options.get('openmp', False):
+            self.cfg.update('configopts', ['--enable-openmp ' + common_config_opts])
+
+        self.log.debug("List of configure options to iterate over: %s", self.cfg['configopts'])
+
+        return super(EB_ELPA, self).run_all_steps(*args, **kwargs)
+
+    def sanity_check_step(self):
+        """Custom sanity check for ELPA."""
+
+        custom_paths = {
+            'dirs': ['lib/pkgconfig', 'bin'],
+        }
+
+        shlib_ext = get_shared_lib_ext()
+
+        extra_files = []
+
+        # ELPA uses the following naming scheme:
+        #  "onenode" suffix: no MPI support
+        #  "openmp" suffix: OpenMP support
+        if self.toolchain.options.get('usempi', None):
+            mpi_suff = ''
+        else:
+            mpi_suff = '_onenode'
+
+        for with_omp in [False, self.toolchain.options.get('openmp', False)]:
+            if with_omp:
+                omp_suff = '_openmp'
+            else:
+                omp_suff = ''
+
+            extra_files.append('include/elpa%s%s-%s/elpa/elpa.h' % (mpi_suff, omp_suff, self.version))
+            extra_files.append('include/elpa%s%s-%s/modules/elpa.mod' % (mpi_suff, omp_suff, self.version))
+
+            extra_files.append('lib/libelpa%s%s.a' % (mpi_suff, omp_suff))
+            if self.cfg['with_shared']:
+                extra_files.append('lib/libelpa%s%s.%s' % (mpi_suff, omp_suff, shlib_ext))
+
+        custom_paths['files'] = nub(extra_files)
```

Comment:
@micaeljtoliveira The `nub` isn't actually needed anymore, is it? I don't see how there can be any duplicates (anymore)?

If so, please remove that (and drop the `import` as well).

It's not a big issue to have it there, but since `nub` has been moved to another place for EasyBuild 4.0, we should get rid of it if it's not actually needed...

Reply:
@boegel Actually it's possible to have duplicates and that's because of a change you requested ;)

The culprit is this line:
`for with_omp in [False, self.toolchain.options.get('openmp', False)]:`

If `openmp = False`, the loop will iterate twice over the value `False`.

Reply:
Oh, right... So we do need `nub`, but I've move it up here:

```python
for with_omp in nub([False, self.toolchain.options.get('openmp', False)]):